pycbc/workflow/grb_utils.py

```diff
@@ -239,6 +239,48 @@ def get_sky_grid_scale(
     return out
 
 
+def generate_tc_prior(wflow, tc_path, buffer_seg):
+    """
+    Generate the configuration file for the prior on the coalescence
+    time of injections, ensuring that these times fall in the analysis
+    time and avoid the onsource and its buffer.
+
+    Parameters
+    ----------
+    tc_path : str
+        Path where the configuration file for the prior needs to be written.
+    buffer_seg : segmentlist
+        Start and end times of the buffer segment encapsulating the onsource.
+    """
+
+    # Write the tc-prior configuration file if it does not exist
+    if os.path.exists(tc_path):
+        raise ValueError("Refusing to overwrite %s." % tc_path)
+    tc_file = open(tc_path, "w")
+    tc_file.write("[prior-tc]\n")
+    tc_file.write("name = uniform\n")
+    tc_file.write("min-tc = %s\n" % wflow.analysis_time[0])
+    tc_file.write("max-tc = %s\n\n" % wflow.analysis_time[1])
+    tc_file.write("[constraint-tc]\n")
+    tc_file.write("name = custom\n")
+    tc_file.write("constraint_arg = (tc < %s) | (tc > %s)\n" %
+                  (buffer_seg[0], buffer_seg[1]))
+    tc_file.close()
+
+    # Add the tc-prior configuration file url to wflow.cp if necessary
+    tc_file_path = "file://"+tc_path
+    for inj_sec in wflow.cp.get_subsections("injections"):
+        config_urls = wflow.cp.get("workflow-injections",
+                                   inj_sec+"-config-files")
+        config_urls = [url.strip() for url in config_urls.split(",")]
+        if tc_file_path not in config_urls:
+            config_urls += [tc_file_path]
+        config_urls = ', '.join([str(item) for item in config_urls])
+        wflow.cp.set("workflow-injections",
```

Comment:
As a general comment, I'm not sure I like the method of "generate new file and then copy it around by editing the ConfigParser object". In my vision for the workflow module the ConfigParser object should be immutable (after directives and shortcuts have been resolved when it's read in).

I would prefer to pass any new files around.

Reply:
Yeah, I'm not fond of that either but I can't think of another solution because `create_injections` does not admit specifying a prior like that from command line, as far as I understand.  PyGRB has already been forced to do something like this for years [here](https://github.com/gwastro/pycbc/blob/master/pycbc/workflow/grb_utils.py#L80), by the way :-/

Reply:
I'm happy using a File to store the prior, but I think it should be passed around the workflow as a File object, not by editing the ini file.

However, that's just my opinion, and I'm happy for you to set standards in the `grb_utils` module, so approval is given if you want to merge as is.